In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('/content/healthcare_disease_prediction_dataset.csv')

In [3]:
df.isnull().sum()



,0
Age,0
Gender,0
Blood Pressure,0
Cholesterol,0
Glucose,0
Smoking,0
Alcohol Consumption,0
Exercise,0
BMI,0
Family History,0


there are no missing values thus i don't have to data-cleaning before training the model.

In [4]:
from sklearn.preprocessing import LabelEncoder, StandardScaler
from sklearn.model_selection import train_test_split

for col in df.select_dtypes(include='object').columns:
    le = LabelEncoder()
    df[col] = le.fit_transform(df[col])

disease_cols = [
    "Heart Disease", "Diabetes", "Stroke", "Kidney Disease",
    "Cancer", "Alzheimer's Disease", "COPD",
    "Liver Disease", "Parkinson's Disease", "Tuberculosis"
]

X = df.drop(disease_cols, axis=1)
y = df['Heart Disease']






removed possible data leakage by dropping disease_cols

In [5]:
X_train, X_test, y_train, y_test = train_test_split(
    X,y, test_size=0.2, random_state=42
)

scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)


split the processed data : 80% to train the model, then the remaining 20% to test the model.

In [7]:
from sklearn.ensemble import RandomForestClassifier
model = RandomForestClassifier()
model.fit(X_train, y_train)


RandomForestClassifier()

In [8]:
model.score(X_test, y_test)

0.705

the model has 70.5% accuracy which is decent but can be fine tuned for higher accurary.